### ISSUE TITLE
Are retries supported?

### ISSUE THREAD

**Issue by maxwnewcomer** [lexiforest/curl_cffi#24]
Just wondering if I'm missing something. Is there a retry and/or backoff factor included in the requests made by this package? 

Is it possible to mount those retries onto a session? With the normal request package it would look like:
```python
session.mount('https://', HTTPAdapter(max_retries=requests.packages.urllib3.util.retry.Retry(**kwargs)))
```

**Comment by yifeikong** [comment]
No, this package does not use urllib3 under the hood, so it does not make sense to use its adapters. Besides, libcurl does not expose an option like CURLOPT_RETRY, so there is no easy fix. But this is a nice feature to have, I'll consider adding it in the future.

If you need it now, you'll have to implement it yourself, [here](https://github.com/encode/httpx/issues/108#issuecomment-1132753155) is a nice place to take some ideas from.
